### PR TITLE
Implemented skip functionality on pipeline page(#23zj8cv)

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/views/Pipeline.vue [hunk 1]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
...
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 10){
+                  const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus({hours: this.temporalExpr(job.tempExprId).integer2});
+                  const timeDiff = job.runTime - DateTime.local().toMillis()
+                  job.runTime = job.runTime + time - timeDiff
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 5){
+                  const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus({days: this.temporalExpr(job.tempExprId).integer2});  
+                  const timeDiff = job.runTime - DateTime.local().toMillis()
+                  job.runTime = job.runTime + time - timeDiff
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:146]: We can optimize this code by writing these three lines outside of the if-else and initializing a variable on the basis of the condition, which will be further used in the `plus` method.
```
let skipTime = '' // outside if-else

skipTime = {days: this.temporalExpr(job.tempExprId).integer2} // on the basis of condition
                  
// outside if-else block
const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus(skipTime);  
const timeDiff = job.runTime - DateTime.local().toMillis()
job.runTime = job.runTime + time - timeDiff
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 1 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Improved the code.

File: src/views/Pipeline.vue [hunk 2]
```diff
@@ -117,6 +117,47 @@ export default defineComponent({
...
+                if(this.temporalExpr(job.tempExprId).integer1 === 12){
+                  skipTime = {minutes: this.temporalExpr(job.tempExprId).integer2}
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 10){
+                  skipTime = {hours: this.temporalExpr(job.tempExprId).integer2}
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 5){
+                  skipTime = {days: this.temporalExpr(job.tempExprId).integer2} 
+                }
+                console.log(skipTime)
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:143]: Missed a console.log

File: src/views/Pipeline.vue [hunk 3]
```diff
@@ -117,6 +117,47 @@ export default defineComponent({
...
+          buttons: [
+            {
+              text: this.$t('Dont skip'),
+              role: 'cancel',
+            },
+            {
+              text: this.$t('Skip'),
+              handler: () => {
+                let skipTime = {};
+                if(this.temporalExpr(job.tempExprId).integer1 === 12){
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:134]: `this.temporalExpr(job.tempExprId).integer1` and `this.temporalExpr(job.tempExprId).integer2` can also be stored in a variable and then used everywhere, WDYT?

File: src/views/Pipeline.vue [hunk 4]
```diff
@@ -117,6 +117,47 @@ export default defineComponent({
...
+                  skipTime = {minutes: this.temporalExpr(job.tempExprId).integer2}
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 10){
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:137]: What if the integer1 value does not match any condition?

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 1 above)
Comment (ymaheshwari1) [reply src/views/Pipeline.vue]: Thank You.

File: src/views/Pipeline.vue [hunk 5]
```diff
@@ -117,6 +117,47 @@ export default defineComponent({
...
       })
     },
+    async skipJob (job: any) {
+      const alert = await alertController
+        .create({
+          header: this.$t('Skip job'),
+          message: this.$t('Skipping will run this job at the next occurance based on the temporal expression.'),
+          buttons: [
+            {
+              text: this.$t('Dont skip'),
```
Comment (dt2patel) [inline src/views/Pipeline.vue:127]: Need to add apostrophe in Don't will need to fix internationalization file too.

```suggestion
              text: this.$t('Don't skip'),
```

**Review by dt2patel — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 2 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Removed.

File: src/views/Pipeline.vue (hunk 3 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Made the requested change.

File: src/views/Pipeline.vue (hunk 5 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Added apostrophe in Don't

File: src/locales/en.json [hunk 6]
```diff
@@ -29,7 +29,7 @@
   "Days": "Days",
   "Disable": "Disable",
   "Discard changes": "Discard changes",
-  "Dont cancel": "Dont cancel",
+  "Don't cancel": "Dont cancel",
```
Comment (adityasharma7) [inline src/locales/en.json:32]:
```suggestion
  "Don't cancel": "Don't cancel",
```

File: src/views/Pipeline.vue [hunk 7]
```diff
@@ -22,23 +22,23 @@
...
-
             <ion-item>
               <ion-icon slot="start" :icon="timeOutline" />
               <ion-label>{{ job.runTime ? getTime(job.runTime) : "-"  }}</ion-label>
             </ion-item>
 
             <ion-item>
               <ion-icon slot="start" :icon="timerOutline" />
-              <ion-label>{{ job.tempExprId ? temporalExpr(job.tempExprId) : "🙃"  }}</ion-label>
+              <!-- {{temporalExpr(job.tempExprId)}} -->
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:32]: I think you missed some commented code

File: src/views/Pipeline.vue [hunk 8]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
         event.target.complete();
       })
     },
+    async skipJob (job: any) {
+      const alert = await alertController
+        .create({
+          header: this.$t('Skip job'),
+          message: this.$t('Skipping will run this job at the next occurance based on the temporal expression.'),
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:124]: I think the text is not added to en.json. Please check for other cases as well
```suggestion
          message: this.$t('Skipping will run this job at the next occurence based on the temporal expression.'),
```

File: src/views/Pipeline.vue [hunk 9]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
...
+              text: this.$t("Don't skip"),
+              role: 'cancel',
+            },
+            {
+              text: this.$t('Skip'),
+              handler: async () => {
+                let skipTime = {};
+                const integer1 = this.temporalExpr(job.tempExprId).integer1;
+                const integer2 = this.temporalExpr(job.tempExprId).integer2
+                if(integer1 === 12) {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:136]: Write if-else in continuity

File: src/views/Pipeline.vue [hunk 10]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
...
+                  skipTime = { minutes: integer2 }
+                }
+                else if (integer1 === 10) {
+                  skipTime = { hours: integer2 }
+                }
+                else if (integer1 === 5) {
+                  skipTime = { days: integer2 }
+                }
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:144]: Handle case if none of the above conditions satisfy

File: src/views/Pipeline.vue [hunk 11]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
...
+                const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus(skipTime);  
+                const timeDiff = job.runTime - DateTime.local().toMillis()
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:146]: I think timeDiff should not be needed. We will set date time based upon last set runtime. Let me know if I am missed something here

File: src/views/Pipeline.vue [hunk 12]
```diff
@@ -117,6 +117,48 @@ export default defineComponent({
...
+                const updatedRunTime = job.runTime + time - timeDiff
+                const payload = {
+                  'jobId': job.jobId,
+                  'runTime': updatedRunTime,
+                  'systemJobEnumId': job.systemJobEnumId,
+                  'statusId': "SERVICE_PENDING"
+                } as any
+                this.store.dispatch('job/updateJob', payload);
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:154]:
```suggestion
                await this.store.dispatch('job/updateJob', payload);
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/locales/en.json (hunk 6 above)
Comment (disha1202) [reply src/locales/en.json]: Made the requested change.

File: src/views/Pipeline.vue (hunk 7 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Removed commented code.

File: src/views/Pipeline.vue (hunk 8 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: The text was already added to en.json file.

File: src/views/Pipeline.vue (hunk 9 above)
Comment (disha1202) [reply src/views/Pipeline.vue:137]: Made the requested change.

File: src/views/Pipeline.vue (hunk 10 above)
Comment (disha1202) [reply src/views/Pipeline.vue:146]: Handled case if none of the above conditions satisfy.

File: src/views/Pipeline.vue (hunk 11 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Yes Sir, updated the code as per your suggestion.

File: src/views/Pipeline.vue (hunk 12 above)
Comment (disha1202) [reply src/views/Pipeline.vue]: Made the requested changes.

File: src/locales/en.json [hunk 13]
```diff
@@ -95,13 +95,14 @@
   "Skip": "Skip",
   "Skip job": "Skip job",
   "Skip once": "Skip once",
-  "Skipping will run this job at the next occurance based on the temporal expression.": "Skipping will run this job at the next occurance based on the temporal expression.",
+  "Skipping will run this job at the next occurence based on the temporal expression.": "Skipping will run this job at the next occurence based on the temporal expression.",
```
Comment (adityasharma7) [inline src/locales/en.json:98]:
```suggestion
  "Skipping will run this job at the next occurrence based on the temporal expression.": "Skipping will run this job at the next occurrence based on the temporal expression.",
```

File: src/views/Pipeline.vue [hunk 14]
```diff
@@ -117,6 +118,48 @@ export default defineComponent({
         event.target.complete();
       })
     },
+    async skipJob (job: any) {
+      const alert = await alertController
+        .create({
+          header: this.$t('Skip job'),
+          message: this.$t('Skipping will run this job at the next occurence based on the temporal expression.'),
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:125]:
```suggestion
          message: this.$t('Skipping will run this job at the next occurrence based on the temporal expression.'),
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**